nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -76,6 +77,17 @@ in {
       default = "/var/lib/nextcloud";
       description = "Storage path of nextcloud.";
     };
+    datadir = mkOption {
+      type = types.str;
+      default = null;
+      description = "Data storage path of nextcloud. Will be services.nextcloud.home by default. This folder will be populated with a config and data folder which contain the state of the instance (excl the database).";
+      example = "/mnt/nextcloud-files";
+    };
+    extraApps = mkOption {
+      type = types.attrsOf types.package;
+      default = { };
+      description = "Extra apps to install. Should point to packages that have appinfo/info.xml in their root.";
```

Comment:
To me it's not entirely clear how the attribute-name has to look like. I assume that `maps = ...` where the repository is also named `maps`, correct? It would be good if that would be documented here. if the description becomes too long, you may want to use multi-line strings btw:

```
description = ''
  very
  long
  description
'';
```

Reply:
done